src/stdlib/SDL_random.c

```diff
@@ -61,14 +61,14 @@ Uint32 SDL_rand(void)
     return (Uint32)(SDL_rand_state >> 32);
 }
 
-Uint32 SDL_rand_n(Uint32 n)
+Sint32 SDL_rand_n(Sint32 n)
 {
 	// On 32-bit arch, the compiler will optimize to a single 32-bit multiply
-	Uint64 val = (Uint64)SDL_rand() * n;
-	return (Uint32)(val >> 32);
+	Uint64 val = (Uint64)SDL_rand_bits() * n;
+	return (Sint32)(val >> 32);
```

Comment:
This needs to shift 33 bits, right?

Reply:
No. It's treating the random number as a 0.32 bit fixed point number. It then multiplies by the 31 bit signed int and gets a 31.32 bit fixed point result. We then shift by 32 bits to get the 31 bit integer we want.

It could probably use a comment.

Reply:
And possibly `if (n <= 0) { return 0; }`?